src/org/labkey/test/pages/dataset/EditDatasetDefinitionPage.java

```diff
@@ -0,0 +1,282 @@
+import org.labkey.test.components.domain.DomainDesigner;
+import org.labkey.test.components.glassLibrary.components.FilteringReactSelect;
+import org.labkey.test.components.glassLibrary.components.ReactSelect;
+import org.labkey.test.components.html.Checkbox;
+import org.labkey.test.components.html.Input;
+import org.labkey.test.components.html.RadioButton;
+import org.labkey.test.pages.DatasetPropertiesPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_PAGE;
+import static org.labkey.test.WebDriverWrapper.waitFor;
+
+public class EditDatasetDefinitionPage extends DomainDesigner<EditDatasetDefinitionPage.ElementCache>
+{
+    public EditDatasetDefinitionPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public static EditDatasetDefinitionPage beginAt(WebDriverWrapper webDriverWrapper)
+    {
+        return beginAt(webDriverWrapper, webDriverWrapper.getCurrentContainerPath());
+    }
+
+    public static EditDatasetDefinitionPage beginAt(WebDriverWrapper webDriverWrapper, String containerPath)
+    {
+        webDriverWrapper.beginAt(WebTestHelper.buildURL("study", containerPath, "defineDatasetType"));
+        return new EditDatasetDefinitionPage(webDriverWrapper.getDriver());
+    }
+
+    public void waitForPage()
+    {
+        waitFor(()-> getFieldsPanel().getComponentElement().isDisplayed(),
+                "The page did not render in time", WAIT_FOR_PAGE);
+    }
+
+    public EditDatasetDefinitionPage shareDemographics(String option)
+    {
+        expandPropertiesPanel();
+        openAdvancedDatasetSettings()
+            .shareDemographics(option)
+            .clickApply();
+        return this;
+    }
+
+    public AdvancedDatasetSettingsDialog openAdvancedDatasetSettings()
+    {
+        expandPropertiesPanel();
+        elementCache().advancedSettingsButton.click();
+        return new AdvancedDatasetSettingsDialog(this);
+    }
+
+    public EditDatasetDefinitionPage setName(String name)
+    {
+        expandPropertiesPanel();
+        elementCache().nameInput.set(name);
+        return this;
+    }
+
+    public String getName()
+    {
+        return elementCache().nameInput.get();
+    }
+
+    public EditDatasetDefinitionPage setDescription(String description)
+    {
+        expandPropertiesPanel();
+        elementCache().descriptionInput.set(description);
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return elementCache().descriptionInput.get();
+    }
+
+    // properties panel interactions
+    // get/set category
+    public EditDatasetDefinitionPage setCategory(String category)
+    {
+        expandPropertiesPanel();
+        elementCache().categorySelect.filterSelect(category);
```

Comment:
Note that this "Category" input is the only dropdown in this panel which also allows for a text input that does not match a dropdown option. This is for the case of allowing a new category to be created for this dataset.

Reply:
Yes, the page puts a FilteringReactSelect there because its 'filterSelect' function supports that scenario.  At least one test sets a category that doesn't exist yet, and it works.